include/oneapi/dpl/pstl/utils.h

```diff
@@ -921,11 +921,11 @@ struct __min_nested_type_size<oneapi::dpl::__internal::tuple<_Ts...>>
     constexpr static std::size_t value = std::min({__min_nested_type_size<_Ts>::value...});
 };
 
-template <typename _ReferenceType1, typename _ReferenceType2>
-struct __swap_ranges_fn
+struct __swap_fn
 {
+    template <typename _Type1, typename _Type2>
     void
-    operator()(_ReferenceType1 __x, _ReferenceType2 __y) const
+    operator()(_Type1&& __x, _Type2&& __y) const
```

Comment:
I believe this change looks not quite correct.
Because now you are using here `universal reference` : this mean you may pass into this function `const reference` too.
But it's invalid use case and I believe better to rewrite this like
```C++
operator()(_Type1& __x, _Type2& __y) const
```

Reply:
> But it's invalid use case and I believe better to rewrite this like

No.... For example, a proxy type, like `tuple<int&, int&>` doesn't match to `_Type1& __x`.

Reply:
Ok